src/ast/passes/codegen_llvm.cpp

```diff
@@ -1926,11 +1926,30 @@ void CodegenLLVM::visit(Cast &cast)
   {
     // If casting to a pointer and the size of the integer expression doesn't
     // match the pointer size (e.g. "(struct Foo*)(uint8)42"), add an explicit
-    // cast.
+    // conversion.
     auto pointer_ty = b_.GetType(cast.type);
     if (expr_->getType() != pointer_ty)
     {
-      expr_ = b_.CreateIntCast(expr_, pointer_ty, false, "cast");
+      // Rather than keeping expr_ as an integer expression and casting it,
```

Comment:
CreateIntCast -> CreateIntToPtr feels like the right call, given that it expresses the semantics better (a pointer cast should return a pointer!).

However, I am not 100% sure I follow why CreateIntCast casts up to i64 given that it's going off of pointer_ty, which should be 32-bit.